linked_list/linked_list.py

```diff
@@ -23,13 +23,70 @@ def __init__(self):
     self.tail = None
 
   def add_to_tail(self, value):
-    pass
+    new = Node(value)
+    if self.tail is None:
+      self.tail  = new
+      self.head = new
+    elif self.head == self.tail:
+      self.head.set_next(new)
+      self.tail = new
```

Comment:
Ah, I think I see what you're trying to do here. What's interesting that if `self.head` == `self.tail`, both `head` & `tail` will have a reference to the same `node` object in memory. Thus, you only need to invoke `set_next()` for `self.tail`; if `self.tail` and `self.head` is the same object, changing `self.tail` changes `self.head` as well, since they point to the same object.